include/aws/s3/s3_client.h

```diff
@@ -206,6 +227,12 @@ struct aws_s3_meta_request_options {
 
     /* Callback for when the meta request has completely cleaned up. */
     aws_s3_meta_request_shutdown_fn *shutdown_callback;
+
+    /**
+     * Invoked to report progress of multi-part upload and copy object requests.
+     * See `aws_s3_meta_request_progress_fn`
+     */
+    aws_s3_meta_request_progress_fn *progress_callback;
```

Comment:
I don't see any where related "multi-part upload" in this PR. Is that something "TODO"? And, what about get object? What if people set this callback for get object request? Do this callback just never be invoked or error out?

I think we could try to support this callback for all the types of meta requests we have. We can add TODO here, if it's something not related to this PR.

Reply:
Good point! Updated the comment to reflect the current behavior and added `TODO` to add support for the other meta request types (to be addressed in distinct CRs). Thanks!